src/sphinxexpr.cpp

```diff
@@ -9169,11 +9397,42 @@ ISphExpr * ExprParser_t::CreateAggregateNode ( const ExprNode_t & tNode, ESphAgg
 	switch ( tLeft.m_iToken )
 	{
 		case TOK_ATTR_JSON:			return new Expr_JsonFieldAggr_c ( pLeft, eFunc );
-		case TOK_ATTR_MVA32:		return new Expr_MVAAggr_c<DWORD> ( tLeft.m_tLocator, GetNameByLocator(tLeft), eFunc );
-		case TOK_ATTR_MVA64:		return new Expr_MVAAggr_c<int64_t> ( tLeft.m_tLocator, GetNameByLocator(tLeft), eFunc );
+		case TOK_ATTR_MVA32:
+		{
+			// Check if attribute is from storage (sorted) or computed expression (unsorted)
+			bool bIsSorted = true;
```

Comment:
it could be better to move all code that set `bIsSorted` into static function and call that function at the  `case TOK_ATTR_MVA32:` and at the `case TOK_ATTR_MVA64:` like this
```
		case TOK_ATTR_MVA32:
		{
			bool bIsSorted = CheckIsStored ( m_pSchema, tLeft );
			return new Expr_MVAAggr_c<DWORD> ( tLeft.m_tLocator, GetNameByLocator(tLeft), eFunc, bIsSorted );
		}
```